### ISSUE TITLE
Error with PHP 8.0 and Composer 2.0.7

### ISSUE THREAD

**Issue by LeRoyJulian** [composer/composer#9522]
Just upgraded to PHP 8.0 using 'brew upgrade', now trying to use composer results in:

```bash
➜  ~ composer global update                                                
PHP Warning:  require_once(phar:///usr/local/bin/composer/vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8): Failed to open stream: phar error: "vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8" is not a file in phar "/usr/local/bin/composer" in phar:///usr/local/bin/composer/vendor/symfony/polyfill-mbstring/bootstrap.php on line 115

Warning: require_once(phar:///usr/local/bin/composer/vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8): Failed to open stream: phar error: "vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8" is not a file in phar "/usr/local/bin/composer" in phar:///usr/local/bin/composer/vendor/symfony/polyfill-mbstring/bootstrap.php on line 115
PHP Fatal error:  Uncaught Error: Failed opening required 'phar:///usr/local/bin/composer/vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8' (include_path='.:/usr/local/Cellar/php/8.0.0/share/php/pear') in phar:///usr/local/bin/composer/vendor/symfony/polyfill-mbstring/bootstrap.php:115
Stack trace:
#0 phar:///usr/local/bin/composer/vendor/composer/autoload_real.php(69): require()
#1 phar:///usr/local/bin/composer/vendor/composer/autoload_real.php(59): composerRequireComposerPhar1603530979('0e6d7bf4a5811bf...', 'phar:///usr/loc...')
#2 phar:///usr/local/bin/composer/vendor/autoload.php(7): ComposerAutoloaderInitComposerPhar1603530979::getLoader()
#3 phar:///usr/local/bin/composer/src/bootstrap.php(15): include('phar:///usr/loc...')
#4 phar:///usr/local/bin/composer/src/bootstrap.php(18): includeIfExists('phar:///usr/loc...')
#5 phar:///usr/local/bin/composer/bin/composer(8): require('phar:///usr/loc...')
#6 /usr/local/bin/composer(24): require('phar:///usr/loc...')
#7 {main}
  thrown in phar:///usr/local/bin/composer/vendor/symfony/polyfill-mbstring/bootstrap.php on line 115

Fatal error: Uncaught Error: Failed opening required 'phar:///usr/local/bin/composer/vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8' (include_path='.:/usr/local/Cellar/php/8.0.0/share/php/pear') in phar:///usr/local/bin/composer/vendor/symfony/polyfill-mbstring/bootstrap.php:115
Stack trace:
#0 phar:///usr/local/bin/composer/vendor/composer/autoload_real.php(69): require()
#1 phar:///usr/local/bin/composer/vendor/composer/autoload_real.php(59): composerRequireComposerPhar1603530979('0e6d7bf4a5811bf...', 'phar:///usr/loc...')
#2 phar:///usr/local/bin/composer/vendor/autoload.php(7): ComposerAutoloaderInitComposerPhar1603530979::getLoader()
#3 phar:///usr/local/bin/composer/src/bootstrap.php(15): include('phar:///usr/loc...')
#4 phar:///usr/local/bin/composer/src/bootstrap.php(18): includeIfExists('phar:///usr/loc...')
#5 phar:///usr/local/bin/composer/bin/composer(8): require('phar:///usr/loc...')
#6 /usr/local/bin/composer(24): require('phar:///usr/loc...')
#7 {main}
  thrown in phar:///usr/local/bin/composer/vendor/symfony/polyfill-mbstring/bootstrap.php on line 115
```

**Comment by stof** [comment]
are you sure that you are using composer 2.0.7 and not an older version of composer 2.0 from before the bug was fixed ?

**Comment by LeRoyJulian** [comment]
Sure. I even reinstalled from source before posting.

**Comment by LeRoyJulian** [comment]
My bad... I didn't updated the global bin. Thanks.

**Comment by Negi47** [comment]
I am getting the same error may I know how @LeRoyJulian  you fixed the problem.

**Comment by sptdigital** [comment]
Whoever bumps into this, I was running composer 2.0.0, if I understood correctly composer 2.0.7 had a fix, I had no reason to stay on a 2.0.0 so I have updated my composer to the latest version (2.7) at the mo and the issue disappeared.

`composer self-update --2`

reported here: [https://github.com/composer/composer/issues/9331](https://github.com/composer/composer/issues/9331)

I should say I could not run composer anything, it would always result in the described error. I am on a multi php versions env where I had possibility to switch to a working php version 7.4 and update composer from there

`sudo update-alternatives --config php`

If you don't have the same setup you'll need to go the composer-setup way specifying the version you want
